tools/Microsoft.AspNetCore.Server.Kestrel.GeneratedCode/KnownHeaders.cs

```diff
@@ -228,13 +231,23 @@ public static string GeneratedFile()
                 "Set-Cookie",
                 "Vary",
                 "WWW-Authenticate",
-            }).Concat(corsResponseHeaders).Select((header, index) => new KnownHeader
+            })
+            .Concat(corsResponseHeaders)
+            .Select((header, index) => new KnownHeader
             {
                 Name = header,
                 Index = index,
                 EnhancedSetter = enhancedHeaders.Contains(header),
                 PrimaryHeader = responsePrimaryHeaders.Contains(header)
-            }).ToArray();
+            })
+            .Concat(new[] { new KnownHeader
+            {
+                Name = "Content-Length",
+                Index = -1,
+                EnhancedSetter = enhancedHeaders.Contains("Content-Length"),
+                PrimaryHeader = responsePrimaryHeaders.Contains("Content-Length")
+            }})
+            .ToArray();
```

Comment:
63 for reponseHeaders as it steals one for Content-Length in `CopyTo(ref MemoryPoolIterator output)`